src/core/scenario.jl

```diff
@@ -644,11 +647,54 @@ function Scenario(d::Dict; flex_hvac_from_json=false)
         end
     end
 
+    # Electric Heater
     electric_heater = nothing
     if haskey(d, "ElectricHeater") && d["ElectricHeater"]["max_mmbtu_per_hour"] > 0.0
         electric_heater = ElectricHeater(;dictkeys_tosymbols(d["ElectricHeater"])...)
     end
 
+    # ASHP
+    ashp = nothing
+    cop_heating = []
+    cop_cooling = []
+    if haskey(d, "ASHP") && d["ASHP"]["max_ton"] > 0.0
+        # If user does not provide heating cop series then assign cop curves based on ambient temperature
+        if !haskey(d["ASHP"], "cop_heating") || !haskey(d["ASHP"], "cop_cooling")
+            # If PV is evaluated, get ambient temperature series from PVWatts and assign PV production factor
+            if isnothing(ambient_temp_celsius)
+                if !isempty(pvs)
+                    for pv in pvs
+                        pv.production_factor_series, ambient_temp_celsius = call_pvwatts_api(site.latitude, site.longitude; tilt=pv.tilt, azimuth=pv.azimuth, module_type=pv.module_type, 
+                            array_type=pv.array_type, losses=round(pv.losses*100, digits=3), dc_ac_ratio=pv.dc_ac_ratio,
+                            gcr=pv.gcr, inv_eff=pv.inv_eff*100, timeframe="hourly", radius=pv.radius, time_steps_per_hour=settings.time_steps_per_hour)
+                    end
+                else
+                    # if PV is not evaluated, call PVWatts to get ambient temperature series
+                    pv_prodfactor, ambient_temp_celsius = call_pvwatts_api(site.latitude, site.longitude; time_steps_per_hour=settings.time_steps_per_hour)    
+                end
+            end
+
+            if !haskey(d["ASHP"], "cop_heating")
+                cop_heating = round.(0.083 .* ambient_temp_celsius .+ 2.8255, digits=2)
```

Comment:
The COPs must have been updated in the document since I coded this in. Updated here https://github.com/NREL/REopt.jl/pull/372/commits/2a77f89edac6d4560b8c7e3d7a3b66bd8e730d77